borg/repository.py

```diff
@@ -376,17 +377,14 @@ def _rebuild_usage(self, segment):
         for tag, key, offset, size in self.io.iter_objects(segment):
             if tag == TAG_PUT:
                 if self.index.get(key, (-1, -1)) != (segment, offset):
+                    # This PUT is superseded in a later segment
```

Comment:
it could be also superseded at a later offset in this segment, but that would not make a difference for freeable space in this segment.